packages/application/src/app/index.ts

```diff
@@ -0,0 +1,221 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+import { createRendermimePlugins } from '@jupyterlab/application/lib/mimerenderers';
+import { PageConfig } from '@jupyterlab/coreutils';
+import { DocumentRegistry } from '@jupyterlab/docregistry';
+import { IRenderMime } from '@jupyterlab/rendermime-interfaces';
+import { IIterator, iter } from '@lumino/algorithm';
+import { Token } from '@lumino/coreutils';
+import { Message } from '@lumino/messaging';
+import { Signal, ISignal } from '@lumino/signaling';
+import { BoxLayout, Widget } from '@lumino/widgets';
+
+/**
+ * The Rise application shell token.
+ */
+export const IRetroShell = new Token<RiseShell>('rise-application:IRiseShell');
+
+export class RiseShell extends Widget implements JupyterFrontEnd.IShell {
+  constructor() {
+    super();
+    this._updated = new Signal(this);
+    this.layout = new BoxLayout();
+    this.id = 'main';
+    this._currentWidget = null;
+  }
+
+  /**
+   * Signal emitted when the shell is updated.
+   */
+  get updated(): ISignal<RiseShell, void> {
+    return this._updated;
+  }
+
+  /**
+   * Activates a widget inside the application shell.
+   *
+   * @param id - The ID of the widget being activated.
+   */
+  activateById(id: string): void {
+    // pass no-op
+  }
+
+  /**
+   * Add a widget to the application shell.
+   *
+   * @param widget - The widget being added.
+   *
+   * @param area - Optional region in the shell into which the widget should
+   * be added.
+   *
+   * @param options - Optional flags the shell might use when opening the
+   * widget, as defined in the `DocumentRegistry`.
+   */
+  add(
+    widget: Widget,
+    area?: string,
+    options?: DocumentRegistry.IOpenOptions
+  ): void {
+    if ((this.layout as BoxLayout).widgets.length > 0) {
+      // pass no-op
+    }
+    BoxLayout.setStretch(widget, 1);
+    (this.layout as BoxLayout).addWidget(widget);
```

Comment:
I believe that the intention was that only custom notebook would get added, but my citation manager reference browser gets added:

![Screenshot from 2022-06-21 21-02-34](https://user-images.githubusercontent.com/5832902/174887640-7fde2851-2d42-434f-93da-de8e6ded5c84.png)

This happens after the following call: [`app.shell.add(referenceBrowser, 'left', { rank: 850 });`](https://github.com/krassowski/jupyterlab-citation-manager/blob/e58dd874fa8718cf8bc7a01cd917d4681d235068/src/index.ts#L1147) in `upyterlab-citation-manager`. You would have noted this with other federated extensions like jupyterlab-git which also add themselves to the shell in the left area, but you likely did not because they failed to activate (e.g. `Plugin '@jupyterlab/git:plugin' failed to activate.`) as they requires tokens which are not provided in this context, whereas jupyterlab-citation-manager is not fussy and has all tokens (except for `INotebookTracker`) optional.

Reply:
I would suggest to add `if (area == 'rise') {` or at the very least `if (area == 'main') {`.